.idea/readme.md

```diff
@@ -0,0 +1,121 @@
+<h2>Local Development & Docker Integration w/ Pycharm</h2>
+
+
+This document contains information regarding ways to use Lean’s Docker image in conjunction with local development in Pycharm.
+
+<h2>Getting Setup</h2>
+
+
+Before anything we need to ensure a few things have been done:
+
+
+
+1. Get [Pycharm Professional](https://www.jetbrains.com/pycharm/)**
+
+2. Get [Docker](https://docs.docker.com/get-docker/):
+    *   Follow the instructions for your Operating System
+    *   New to Docker? Try docker getting-started
+3. Pull Lean’s latest image from a terminal
+    *   _docker pull quantconnect/lean_
+
+     
+4. Clone LEAN Locally
+    *   git clone[ https://github.com/QuantConnect/Lean.git](https://github.com/QuantConnect/Lean.git)
+
+5. Open the folder in Pycharm
+
+**PyCharm’s remote debugger requires PyCharm Professional.
+
+<h2>Develop Algorithms Locally, Run in Container</h2>
+
+
+We have set up a relatively easy way to develop algorithms in your local IDE and push them into the container to be run.
+
+Before we can use this method with Windows or Mac OS we need to share the Lean directory with Docker.
+
+<h3>Activate File Sharing for Docker:</h3>
+
+
+
+
+    *   Windows
+        1. [https://docs.docker.com/docker-for-windows/#file-sharing](https://docs.docker.com/docker-for-windows/#file-sharing)
+        2. Share the LEAN root directory with docker
+    *   Mac
+        3. [https://docs.docker.com/docker-for-mac/#file-sharing](https://docs.docker.com/docker-for-mac/#file-sharing)
+        4. Share the LEAN root directory with docker
+
+
+<h3>Lean Configuration</h3>
+
+
+Next we need to be sure that our Lean configuration at .\Launcher\config.json is properly set. Just like running lean locally, the config must reflect what we want Lean to run. In the case of Pycharm we will only be using python algorithms so follow the steps below to ensure your algorithms will run in the docker container. 
+
+You configuration file should look something like this:
+
+<h4>Python:</h4>
+
+
+"algorithm-type-name": "_~AlgorithmName~_",
+
+"algorithm-language": "Python",
+
+"algorithm-location": "../../../Algorithm.Python/_~AlgorithmName~_.py",
+
+<h4>Note About Python Algorithm Location</h4>
+
+
+Our specific configuration binds the Algorithm.Python directory to the container so any algorithm you would like to run must be in that directory. Please ensure your algorithm location looks just the same as the example above.
+
+<h3>Running Lean in the Container</h3>
+
+
+Now that your configuration is all set and your algorithm in the correct location use the run_docker.bat/.sh script located in Lean’s root directory. The script will kick off the process of starting the container.
```

Comment:
nit - maybe can suggest using the PyCharm terminal

Reply:
Done!